datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/timeline/mappers/SchemaFieldBlameMapper.java

```diff
@@ -0,0 +1,188 @@
+        schemaFieldBlame.setSchemaFieldChange(schemaFieldChange);
+
+        schemaBlameMap.put(schemaUrn, schemaFieldBlame);
+      }
+    }
+
+    result.setSchemaFieldBlameList(schemaBlameMap.values()
+        .stream()
+        .filter(schemaFieldBlame -> !schemaFieldBlame.getSchemaFieldChange()
+            .getChangeType()
+            .equals(ChangeOperationType.REMOVE))
+        .collect(Collectors.toList()));
+    result.setSemanticVersionList(semanticVersionStructList);
+    return result;
+  }
+
+  private static Optional<Pair<SemanticVersion, ChangeTransaction>> semanticVersionChangeTransactionPair(
+      ChangeTransaction changeTransaction) {
+    Optional<SemanticVersion> semanticVersion = createSemanticVersion(changeTransaction.getSemVer());
+    return semanticVersion.map(version -> Pair.of(version, changeTransaction));
+  }
+
+  private static Optional<SemanticVersion> createSemanticVersion(String semanticVersionString) {
+    String truncatedSemanticVersion = truncateSemanticVersion(semanticVersionString);
+    try {
+      SemanticVersion semanticVersion = SemanticVersion.parse(truncatedSemanticVersion);
+      return Optional.of(semanticVersion);
+    } catch (SemanticVersion.SemanticVersionParseException e) {
+      return Optional.empty();
+    }
+  }
+
+  // The SemanticVersion is currently returned from the ChangeTransactions in the format "x.y.z-computed". This function
+  // removes the suffix "computed".
+  private static String truncateSemanticVersion(String semanticVersion) {
+    String suffix = "-computed";
+    return semanticVersion.endsWith(suffix) ? semanticVersion.substring(0, semanticVersion.lastIndexOf(suffix))
+        : semanticVersion;
+  }
+
+  private static SchemaFieldChange getLastSchemaFieldChange(ChangeEvent changeEvent, long timestamp, String actor,
+      String semanticVersion, String versionStamp) {
+    SchemaFieldChange schemaFieldChange = new SchemaFieldChange();
+    schemaFieldChange.setTimestampMillis(timestamp);
+    schemaFieldChange.setActor(actor);
+    schemaFieldChange.setSemanticVersion(truncateSemanticVersion(semanticVersion));
+    schemaFieldChange.setSemanticVersionChange(
+        SemanticVersionChangeType.valueOf(SemanticVersionChangeType.class, changeEvent.getSemVerChange().toString()));
+    schemaFieldChange.setChangeType(
+        ChangeOperationType.valueOf(ChangeOperationType.class, changeEvent.getOperation().toString()));
+    schemaFieldChange.setVersionStamp(versionStamp);
+
+    String translatedChangeOperationType;
+    switch (changeEvent.getOperation()) {
+      case ADD:
+        translatedChangeOperationType = "Added";
+        break;
+      case MODIFY:
+        translatedChangeOperationType = "Modified";
+        break;
+      case REMOVE:
+        translatedChangeOperationType = "Removed";
+        break;
+      default:
+        translatedChangeOperationType = "Unknown change made";
```

Comment:
small thing: maybe log a WARN saying that we found something we don't recognize.

Reply:
Done.